lab_3_keywords_textrank/main.py

```diff
@@ -306,11 +290,11 @@ def is_incidental(self, vertex1: int, vertex2: int) -> int:
                 1 if vertices are incidental, otherwise 0
         If either of vertices is not present in the graph, -1 is returned
         """
-        if vertex1 not in self._vertices or vertex2 not in self._vertices:
+        if not (vertex1 in self._vertices and vertex2 in self._vertices):
             return -1
-        idx1 = self._vertices.index(vertex1)
-        idx2 = self._vertices.index(vertex2)
-        return self._matrix[idx1][idx2]
+        if self._matrix[self._vertices.index(vertex1)][self._vertices.index(vertex2)] == 1:
```

Comment:
consider directly returning the value at the intersection of these two vertices: it is already either 0 or 1